cns/restserver/ipam.go

```diff
@@ -1068,6 +1073,14 @@ func (service *HTTPRestService) GetEndpointHelper(endpointID string) (*EndpointI
 		logger.Warnf("[GetEndpointState] Found existing endpoint state for container %s", endpointID)
 		return endpointInfo, nil
 	}
+	// This part is a temprory fix if we have endpoint states belong to CNI version 1.4.X on Windows since the states don't have the containerID
```

Comment:
SO the issue is within CNI 1.4.X and the only info it gioves us is the [first-8-containerID-digits]-eth0 and also the IP address. So the data is not there at CNS to convert them. We need this special case for 1.4.x!

Reply:
the way I previously handled this was via the `Key()` abstraction on the `PodInfo`, so that the business logic didn't actually need to know that there were different identifier schemas in the backend. Would something like that work here?

Reply:
We can add a Key function but then this special case should be implemented there as well. Since stateless CNI will always query via full containerID and we have to build this special case from it.